zerver/lib/user_groups.py

```diff
@@ -175,6 +175,10 @@ def get_user_group_member_ids(user_group: UserGroup, *, recursive: bool = True)
     return list(member_ids)
 
 
+def is_user_in_group(user_group: UserGroup, user: UserProfile, *, recursive: bool = True) -> bool:
+    return user.id in get_user_group_member_ids(user_group, recursive=recursive)
```

Comment:
I think we'll want to do a more efficient query for this operation, if possible.

Reply:
I updated the query for this as below -
```python
def is_user_in_group(user_group: UserGroup, user: UserProfile, *, recursive: bool = True) -> bool:
    if not recursive:
        return UserGroupMembership.objects.filter(user_group=user_group, user_profile=user).exists()

    return get_recursive_group_members(user_group=user_group).filter(id=user.id).exists()
```

I am not using `get_user_group_member_ids`, since `get_user_group_direct_members` returns list and it is used in other places so did not want to change it.

Reply:
That works.